Nos/Views/ReportMenu.swift

```diff
@@ -116,30 +122,90 @@ struct ReportMenuModifier: ViewModifier {
             }
         }
     }
+    
+    typealias TopLevelDisplayName = String
+    /// An enum to simplify the user selection through the sequence of connected
+    /// dialogs
+    enum UserSelection: Equatable {
```

Comment:
I think we will need to split this eventually because we have one menu for reporting notes and another for reporting users. Maybe UserSelection could turn into a `protocol` like `ReportMenuItems` and then could have two implementations like `ReportContentMenuItems`and `ReportAuthorMenuItems` or similar. Or we could somehow pass the ReportTarget in each case and branch on it... that seems messier though.